cache/common.go

```diff
@@ -28,3 +31,40 @@ func checksumOfFile(path string) (string, error) {
 
 	return hex.EncodeToString(hash.Sum(nil)), nil
 }
+
+type Feature string
+
+const (
+	Multithreading    Feature = "MULTITHREADING"
+	MultipartUploader Feature = "MULTIPART_UPLOADER"
+)
+
+func isValidFeature(f string) bool {
+	switch Feature(f) {
+	case Multithreading, MultipartUploader:
+		return true
+	default:
+		return false
+	}
+}
+
+func parseFeatureFlags(flagStr string, logger log.Logger) map[Feature]bool {
```

Comment:
Feature flag handling should not be in a cache specific package. Should be a separate package.